crowbar_framework/app/constraints/api_constraint.rb

```diff
@@ -15,13 +15,25 @@
 #
 
 class ApiConstraint
-  attr_reader :version
+  attr_reader :versions
 
-  def initialize(options)
-    @version = options.fetch(:version)
+  def initialize(*versions)
+    @versions = versions
   end
 
   def matches?(request)
-    request.headers.fetch(:accept) == "application/vnd.crowbar.v#{version}+json"
+    allowed = false
+    major, minor = nil
+    versions.each do |version|
+      major, minor = version.to_s.split(".").map(&:to_i)
```

Comment:
This splitting should really go into the `initialize`.

```ruby
@versions = versions.map{|v| v.to_s.split(".").map(&:to_i)}
```

and then in `matches?`:

```ruby
versions.any? do |major, minor|
  versions_requested = version_mime.match(request.accept)
  versions_requested[:major].to_i == major && versions_requested[:minor].to_i <= minor
end
```